src/document/utils/assert-doc.js

```diff
@@ -34,4 +45,61 @@ const assertDocArray =
         }
       };
 
-export { assertDoc, assertDocArray };
+const assertDocFillParts =
+  process.env.NODE_ENV === "production"
+    ? noop
+    : /**
+       * @param {Doc[]} parts
+       */
+      function (parts) {
+        assertDocArray(parts);
+        if (parts.length > 1 && cleanDoc(parts.at(-1)) === "") {
+          // stripTrailingHardline can transform trailing hardline into empty string.
+          // The trailing empty string is not a problem even if it's a line element.
+          parts = parts.slice(0, -1);
+        }
+        for (const [i, doc] of parts.entries()) {
+          if (i % 2 === 0) {
+            continue;
+          }
+          if (!isValidSeparator(doc)) {
+            const type = getDocType(doc);
+            throw new Error(
+              `Unexpected non-line-break doc at ${i}. Doc type is ${type}.`,
+            );
```

Comment:
Using `InvalidDocError` is not straightforward because `InvalidDocError` considers such documents valid. Error message gets weird and very hard to understand (message: "doc is valid").
I use just Error here.